### PR TITLE
feat: single product page added 

### PR DESCRIPTION
**Pull request by dev-ashishsethi**
image
Files to check:
singleProduct.jsx

### CODE REVIEW

File: src/pages/SingleProduct/singleProduct.jsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  };
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get("/api/products");
+      setProducts(
+        response.data.products.find((products) => products._id === productID)
+      );
+    })();
+  }, []);
+
```
Comment (m26unkwn) [inline src/pages/SingleProduct/singleProduct.jsx:84]: ABSTRACT THIS FROM YOUR COMPONENT

File: src/pages/ProductListing/ProductListing.jsx [hunk 2]
```diff
@@ -332,17 +332,17 @@ export function ProductListing() {
                 }
               />
             )}
+            <Link to={`/product/${items._id}`} className="link-on-card">
+              <img src={items.image} alt="" className="ecommerce-image" />
```
Comment (m26unkwn) [inline src/pages/ProductListing/ProductListing.jsx:337]: DESTRUCTURE ALL THE DATA FROM ITEMS AND THAN USE IT.

**Review by m26unkwn — APPROVED** [review]
CSS is well-written variables that can be used for repeated values.

you can check for the comments otherwise good to merge.

File: src/pages/ProductListing/ProductListing.jsx [hunk 3]
```diff
@@ -332,17 +332,17 @@ export function ProductListing() {
...
```
Comment (PuneetP16) [inline src/pages/ProductListing/ProductListing.jsx:336]: Add alt values for images.

File: src/pages/ProductListing/productListing.css [hunk 4]
```diff
@@ -1,173 +1,174 @@
...
+  grid-template-areas: "sidebar pr";
+  grid-template-columns: 0.2fr 1.5fr;
+  gap: 1rem;
+}
+.link-on-card {
+  background-color: transparent;
+  text-align: center;
+}
+.sidebar {
+  margin-left: 1rem;
```
Comment (PuneetP16) [inline src/pages/ProductListing/productListing.css:15]: format the css over here

File: src/pages/ProductListing/productListing.css [hunk 5]
```diff
@@ -1,173 +1,174 @@
...
+  display: flex;
+  background-color: transparent;
+  justify-content: space-between;
+}
+
+.product-btn-section {
+  display: block;
+  margin-left: 1rem;
+  margin-top: 1rem;
+  /* width: 16rem; */
```
Comment (PuneetP16) [inline src/pages/ProductListing/productListing.css:151]: remove this comment.

File: src/pages/SingleProduct/singleProduct.jsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    console.log("items", items);
+    (async () => {
+      try {
+        const response = await axios({
+          method: "POST",
+          url: "/api/user/wishlist",
+          data: { product: items },
+          headers: { authorization: token },
+        });
+        console.log(response.data);
```
Comment (PuneetP16) [inline src/pages/SingleProduct/singleProduct.jsx:45]: remove console.log before production.

**Review by PuneetP16 — APPROVED** [review]
Awesome work on Product Page. Specially the images of the product.

**Pull request merged**